fuel-merkle/src/binary/merkle_tree.rs

```diff
@@ -37,6 +37,75 @@ pub struct MerkleTree<TableType, StorageType> {
     phantom_table: PhantomData<TableType>,
 }
 
+impl<TableType, StorageType> MerkleTree<TableType, StorageType> {
```

Comment:
I've refactored storage-independent parts of the BMT interface into a separate impl here. Other than `leaves_count`, these changes come from cut + pasting the existing methods into this impl.